posthog/management/commands/backfill_distinct_id_overrides.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+
+import structlog
+from django.core.management.base import BaseCommand, CommandError
+
+from posthog.clickhouse.client.execute import sync_execute
+from posthog.models.team.team import Team
+
+
+logger = structlog.get_logger(__name__)
+
+
+@dataclass
+class BackfillQuery:
+    team_id: int
+
+    def execute(self, dry_run: bool = False) -> None:
+        query = """
+            SELECT
+                team_id,
+                distinct_id,
+                argMax(person_id, version),
+                argMax(is_deleted, version),
+                max(version)
+            FROM person_distinct_id2
+            WHERE
+                team_id = %(team_id)s
+                AND version > 0
+            GROUP BY team_id, distinct_id
```

Comment:
It could (though it could also time out _without_ it as well, even though the `GROUP BY` does make it more likely.) I ran the `SELECT` without `INSERT` on some of the larger teams and didn't run into any issues, but past performance not indicative of future results, etc. If we run into any issues, it's easy to remove.